java-frontend/src/main/java/org/sonar/java/ast/JavaAstScanner.java

```diff
@@ -123,27 +132,21 @@ public void setVisitorBridge(VisitorsBridge visitor) {
 
   @VisibleForTesting
   public static void scanSingleFileForTests(File file, VisitorsBridge visitorsBridge) {
-    scanSingleFileForTests(file, visitorsBridge, new JavaConfiguration(Charset.forName("UTF-8")));
+    scanSingleFileForTests(file, visitorsBridge, new JavaVersionImpl());
   }
 
   @VisibleForTesting
-  public static void scanSingleFileForTests(File file, VisitorsBridge visitorsBridge, JavaConfiguration conf) {
+  public static void scanSingleFileForTests(File file, VisitorsBridge visitorsBridge, JavaVersion javaVersion) {
     if (!file.isFile()) {
       throw new IllegalArgumentException("File '" + file + "' not found.");
     }
-    JavaAstScanner scanner = create(conf, visitorsBridge);
-
-    scanner.scan(Collections.singleton(file));
-  }
-
-  private static JavaAstScanner create(JavaConfiguration conf, @Nullable VisitorsBridge visitorsBridge) {
-    JavaAstScanner astScanner = new JavaAstScanner(JavaParser.createParser(conf.getCharset()));
+    JavaAstScanner astScanner = new JavaAstScanner(JavaParser.createParser(), null);
     if (visitorsBridge != null) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Change this condition so that it does not always evaluate to "true" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2583)